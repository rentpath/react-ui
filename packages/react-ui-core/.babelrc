{
  "env": {
    "production": {
      "presets": [
        "env",
        "react"
      ],
      "plugins": [
        "transform-runtime",
        "transform-decorators-legacy",
        "transform-class-properties",
        "babel-plugin-transform-object-rest-spread",
        "dynamic-import-webpack"
      ]
    },
    "test": {
      "presets":[
        "env",
        "react",
        "stage-0"
      ],
      "plugins": [
        "transform-es2015-modules-commonjs",
        "transform-decorators-legacy",
        "dynamic-import-node"
      ]
    }
  }
}
