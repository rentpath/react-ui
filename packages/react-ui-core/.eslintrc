{
  "parser": "babel-eslint",
  "extends": [
    "airbnb-base",
    "plugin:flowtype/recommended",
  ],
  "rules": {
    "semi": [2, "never"],
    "arrow-parens": [2, "as-needed"],
    "no-underscore-dangle": [0],
    "import/no-extraneous-dependencies": ["error", {
      "devDependencies": true,
      "packageDir": "./package.json"
    }],
    "react/jsx-uses-react": 1,
    "react/jsx-uses-vars": 1,
  },
  "plugins": [
    "react",
    "flowtype",
    "jest"
  ],
  "settings": {
    "import/resolver": {
      "node": {
        "moduleDirectory": [
          "node_modules",
          "packages"
        ]
      }
    },
    "settings": {
      "flowtype": {
        "onlyFilesWithFlowAnnotation": true
      }
    }
  },
  "env": {
    "jest/globals": true,
    "browser": true,
    "node": true
  }
}
