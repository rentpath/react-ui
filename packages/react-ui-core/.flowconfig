[ignore]
./lib
./node_modules
./test

[include]
./src

[libs]

[options]
esproposal.decorators=ignore

[lints]
