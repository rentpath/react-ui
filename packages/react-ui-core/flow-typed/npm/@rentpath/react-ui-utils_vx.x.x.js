// flow-typed signature: a137d1bccb9d876def54c58cc5b6e8c1
// flow-typed version: <<STUB>>/@rentpath/react-ui-utils_v0.0.1/flow_v0.49.1

/**
 * This is an autogenerated libdef stub for:
 *
 *   '@rentpath/react-ui-utils'
 *
 * Fill this stub out by replacing all the `any` types.
 *
 * Once filled out, we encourage you to share your work with the
 * community by sending a pull request to:
 * https://github.com/flowtype/flow-typed
 */

declare module '@rentpath/react-ui-utils' {
  declare module.exports: any;
}

/**
 * We include stubs for each file inside this npm package in case you need to
 * require those files directly. Feel free to delete any files that aren't
 * needed.
 */
declare module '@rentpath/react-ui-utils/lib/composeProps' {
  declare module.exports: any;
}

declare module '@rentpath/react-ui-utils/lib/index' {
  declare module.exports: any;
}

declare module '@rentpath/react-ui-utils/lib/keyMirror' {
  declare module.exports: any;
}

declare module '@rentpath/react-ui-utils/lib/parseArgs' {
  declare module.exports: any;
}

declare module '@rentpath/react-ui-utils/lib/randomId' {
  declare module.exports: any;
}

declare module '@rentpath/react-ui-utils/src/__tests__/composeProps-test' {
  declare module.exports: any;
}

declare module '@rentpath/react-ui-utils/src/__tests__/keyMirror-test' {
  declare module.exports: any;
}

declare module '@rentpath/react-ui-utils/src/__tests__/parseArgs-test' {
  declare module.exports: any;
}

declare module '@rentpath/react-ui-utils/src/__tests__/randomId-test' {
  declare module.exports: any;
}

declare module '@rentpath/react-ui-utils/src/composeProps' {
  declare module.exports: any;
}

declare module '@rentpath/react-ui-utils/src/index' {
  declare module.exports: any;
}

declare module '@rentpath/react-ui-utils/src/keyMirror' {
  declare module.exports: any;
}

declare module '@rentpath/react-ui-utils/src/parseArgs' {
  declare module.exports: any;
}

declare module '@rentpath/react-ui-utils/src/randomId' {
  declare module.exports: any;
}

declare module '@rentpath/react-ui-utils/test/setup' {
  declare module.exports: any;
}

// Filename aliases
declare module '@rentpath/react-ui-utils/lib/composeProps.js' {
  declare module.exports: $Exports<'@rentpath/react-ui-utils/lib/composeProps'>;
}
declare module '@rentpath/react-ui-utils/lib/index.js' {
  declare module.exports: $Exports<'@rentpath/react-ui-utils/lib/index'>;
}
declare module '@rentpath/react-ui-utils/lib/keyMirror.js' {
  declare module.exports: $Exports<'@rentpath/react-ui-utils/lib/keyMirror'>;
}
declare module '@rentpath/react-ui-utils/lib/parseArgs.js' {
  declare module.exports: $Exports<'@rentpath/react-ui-utils/lib/parseArgs'>;
}
declare module '@rentpath/react-ui-utils/lib/randomId.js' {
  declare module.exports: $Exports<'@rentpath/react-ui-utils/lib/randomId'>;
}
declare module '@rentpath/react-ui-utils/src/__tests__/composeProps-test.js' {
  declare module.exports: $Exports<'@rentpath/react-ui-utils/src/__tests__/composeProps-test'>;
}
declare module '@rentpath/react-ui-utils/src/__tests__/keyMirror-test.js' {
  declare module.exports: $Exports<'@rentpath/react-ui-utils/src/__tests__/keyMirror-test'>;
}
declare module '@rentpath/react-ui-utils/src/__tests__/parseArgs-test.js' {
  declare module.exports: $Exports<'@rentpath/react-ui-utils/src/__tests__/parseArgs-test'>;
}
declare module '@rentpath/react-ui-utils/src/__tests__/randomId-test.js' {
  declare module.exports: $Exports<'@rentpath/react-ui-utils/src/__tests__/randomId-test'>;
}
declare module '@rentpath/react-ui-utils/src/composeProps.js' {
  declare module.exports: $Exports<'@rentpath/react-ui-utils/src/composeProps'>;
}
declare module '@rentpath/react-ui-utils/src/index.js' {
  declare module.exports: $Exports<'@rentpath/react-ui-utils/src/index'>;
}
declare module '@rentpath/react-ui-utils/src/keyMirror.js' {
  declare module.exports: $Exports<'@rentpath/react-ui-utils/src/keyMirror'>;
}
declare module '@rentpath/react-ui-utils/src/parseArgs.js' {
  declare module.exports: $Exports<'@rentpath/react-ui-utils/src/parseArgs'>;
}
declare module '@rentpath/react-ui-utils/src/randomId.js' {
  declare module.exports: $Exports<'@rentpath/react-ui-utils/src/randomId'>;
}
declare module '@rentpath/react-ui-utils/test/setup.js' {
  declare module.exports: $Exports<'@rentpath/react-ui-utils/test/setup'>;
}
