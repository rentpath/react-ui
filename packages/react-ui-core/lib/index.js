'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _Button = require('./Button');

Object.defineProperty(exports, 'Button', {
  enumerable: true,
  get: function get() {
    return _Button.Button;
  }
});
Object.defineProperty(exports, 'ToggleButton', {
  enumerable: true,
  get: function get() {
    return _Button.ToggleButton;
  }
});

var _Text = require('./Text');

Object.defineProperty(exports, 'Text', {
  enumerable: true,
  get: function get() {
    return _Text.Text;
  }
});

var _Form = require('./Form');

Object.defineProperty(exports, 'Form', {
  enumerable: true,
  get: function get() {
    return _Form.Form;
  }
});
Object.defineProperty(exports, 'Label', {
  enumerable: true,
  get: function get() {
    return _Form.Label;
  }
});
Object.defineProperty(exports, 'Field', {
  enumerable: true,
  get: function get() {
    return _Form.Field;
  }
});
Object.defineProperty(exports, 'Input', {
  enumerable: true,
  get: function get() {
    return _Form.Input;
  }
});
Object.defineProperty(exports, 'Textarea', {
  enumerable: true,
  get: function get() {
    return _Form.Textarea;
  }
});
Object.defineProperty(exports, 'Select', {
  enumerable: true,
  get: function get() {
    return _Form.Select;
  }
});
Object.defineProperty(exports, 'RangeSlider', {
  enumerable: true,
  get: function get() {
    return _Form.RangeSlider;
  }
});
Object.defineProperty(exports, 'FieldSet', {
  enumerable: true,
  get: function get() {
    return _Form.FieldSet;
  }
});
Object.defineProperty(exports, 'RadioButton', {
  enumerable: true,
  get: function get() {
    return _Form.RadioButton;
  }
});
Object.defineProperty(exports, 'RadioGroup', {
  enumerable: true,
  get: function get() {
    return _Form.RadioGroup;
  }
});

var _Modal = require('./Modal');

Object.defineProperty(exports, 'Modal', {
  enumerable: true,
  get: function get() {
    return _Modal.Modal;
  }
});
Object.defineProperty(exports, 'ModalBody', {
  enumerable: true,
  get: function get() {
    return _Modal.ModalBody;
  }
});
Object.defineProperty(exports, 'Overlay', {
  enumerable: true,
  get: function get() {
    return _Modal.Overlay;
  }
});

var _Collapsible = require('./Collapsible');

Object.defineProperty(exports, 'Collapsible', {
  enumerable: true,
  get: function get() {
    return _Collapsible.Collapsible;
  }
});

var _Counter = require('./Counter');

Object.defineProperty(exports, 'Counter', {
  enumerable: true,
  get: function get() {
    return _Counter.Counter;
  }
});

var _Ratings = require('./Ratings');

Object.defineProperty(exports, 'RatingBar', {
  enumerable: true,
  get: function get() {
    return _Ratings.RatingBar;
  }
});

var _List = require('./List');

Object.defineProperty(exports, 'List', {
  enumerable: true,
  get: function get() {
    return _List.List;
  }
});

var _Card = require('./Card');

Object.defineProperty(exports, 'Card', {
  enumerable: true,
  get: function get() {
    return _Card.Card;
  }
});

var _Title = require('./Title');

Object.defineProperty(exports, 'Title', {
  enumerable: true,
  get: function get() {
    return _Title.Title;
  }
});

var _Dropdown = require('./Dropdown');

Object.defineProperty(exports, 'Dropdown', {
  enumerable: true,
  get: function get() {
    return _Dropdown.Dropdown;
  }
});

var _Menu = require('./Menu');

Object.defineProperty(exports, 'Menu', {
  enumerable: true,
  get: function get() {
    return _Menu.Menu;
  }
});
Object.defineProperty(exports, 'DropdownMenu', {
  enumerable: true,
  get: function get() {
    return _Menu.DropdownMenu;
  }
});

var _AutoSuggestField = require('./AutoSuggestField');

Object.defineProperty(exports, 'AutoSuggestField', {
  enumerable: true,
  get: function get() {
    return _AutoSuggestField.AutoSuggestField;
  }
});

var _LeadForm = require('./LeadForm');

Object.defineProperty(exports, 'LeadForm', {
  enumerable: true,
  get: function get() {
    return _LeadForm.LeadForm;
  }
});
Object.defineProperty(exports, 'Name', {
  enumerable: true,
  get: function get() {
    return _LeadForm.Name;
  }
});
Object.defineProperty(exports, 'Email', {
  enumerable: true,
  get: function get() {
    return _LeadForm.Email;
  }
});
Object.defineProperty(exports, 'Phone', {
  enumerable: true,
  get: function get() {
    return _LeadForm.Phone;
  }
});
Object.defineProperty(exports, 'Message', {
  enumerable: true,
  get: function get() {
    return _LeadForm.Message;
  }
});
Object.defineProperty(exports, 'OptInNewsLetter', {
  enumerable: true,
  get: function get() {
    return _LeadForm.OptInNewsLetter;
  }
});
Object.defineProperty(exports, 'TermsOfService', {
  enumerable: true,
  get: function get() {
    return _LeadForm.TermsOfService;
  }
});

var _LeadModal = require('./LeadModal');

Object.defineProperty(exports, 'LeadModal', {
  enumerable: true,
  get: function get() {
    return _LeadModal.LeadModal;
  }
});

var _Highlighter = require('./Highlighter');

Object.defineProperty(exports, 'Highlighter', {
  enumerable: true,
  get: function get() {
    return _Highlighter.Highlighter;
  }
});

var _Carousel = require('./Carousel');

Object.defineProperty(exports, 'Carousel', {
  enumerable: true,
  get: function get() {
    return _Carousel.Carousel;
  }
});
Object.defineProperty(exports, 'PhotoCarousel', {
  enumerable: true,
  get: function get() {
    return _Carousel.PhotoCarousel;
  }
});

var _Listing = require('./Listing');

Object.defineProperty(exports, 'ListingCell', {
  enumerable: true,
  get: function get() {
    return _Listing.ListingCell;
  }
});
Object.defineProperty(exports, 'ListingComponents', {
  enumerable: true,
  get: function get() {
    return _Listing.ListingComponents;
  }
});

var _Schema = require('./Schema');

Object.defineProperty(exports, 'Schema', {
  enumerable: true,
  get: function get() {
    return _Schema.Schema;
  }
});

var _Drawer = require('./Drawer');

Object.defineProperty(exports, 'Drawer', {
  enumerable: true,
  get: function get() {
    return _Drawer.Drawer;
  }
});

var _DateInput = require('./DateInput');

Object.defineProperty(exports, 'DateInput', {
  enumerable: true,
  get: function get() {
    return _DateInput.DateInput;
  }
});

var _Gmap = require('./Gmap');

Object.defineProperty(exports, 'Gmap', {
  enumerable: true,
  get: function get() {
    return _Gmap.Gmap;
  }
});
Object.defineProperty(exports, 'GmapSpinner', {
  enumerable: true,
  get: function get() {
    return _Gmap.GmapSpinner;
  }
});
Object.defineProperty(exports, 'GmapInteraction', {
  enumerable: true,
  get: function get() {
    return _Gmap.GmapInteraction;
  }
});
Object.defineProperty(exports, 'InfoWindow', {
  enumerable: true,
  get: function get() {
    return _Gmap.InfoWindow;
  }
});
Object.defineProperty(exports, 'Layer', {
  enumerable: true,
  get: function get() {
    return _Gmap.Layer;
  }
});
Object.defineProperty(exports, 'Marker', {
  enumerable: true,
  get: function get() {
    return _Gmap.Marker;
  }
});
Object.defineProperty(exports, 'Markers', {
  enumerable: true,
  get: function get() {
    return _Gmap.Markers;
  }
});
Object.defineProperty(exports, 'OverlayView', {
  enumerable: true,
  get: function get() {
    return _Gmap.OverlayView;
  }
});
Object.defineProperty(exports, 'FreeDrawLayer', {
  enumerable: true,
  get: function get() {
    return _Gmap.FreeDrawLayer;
  }
});

var _ModalStack = require('./ModalStack');

Object.defineProperty(exports, 'ModalStack', {
  enumerable: true,
  get: function get() {
    return _ModalStack.ModalStack;
  }
});