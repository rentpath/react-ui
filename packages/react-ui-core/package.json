{
  "name": "@rentpath/react-ui-core",
  "version": "12.1.0",
  "main": "lib/index.js",
  "license": "MIT",
  "scripts": {
    "clean": "rm -rf lib",
    "build": "NODE_ENV=production babel --ignore __tests__ -d lib src",
    "build:watch": "yarn run build -- --watch",
    "test": "jest",
    "test:watch": "yarn run test -- --watch",
    "lint": "eslint src",
    "lint:fix": "yarn run lint -- --fix"
  },
  "jest": {
    "setupFiles": [
      "<rootDir>/test/shim.js",
      "<rootDir>/test/setup.js"
    ],
    "moduleNameMapper": {
      "\\.(css)$": "<rootDir>/node_modules/jest-css-modules"
    },
    "testPathIgnorePatterns": [
      "/node_modules/",
      "/mocks/"
    ]
  },
  "devDependencies": {
    "@rentpath/eslint-config-rentpath": "3.0.1",
    "babel-cli": "6.26.0",
    "babel-core": "6.26.0",
    "babel-eslint": "^8.2.2",
    "babel-jest": "^22.4.3",
    "babel-plugin-dynamic-import-node": "^1.2.0",
    "babel-plugin-dynamic-import-webpack": "^1.0.2",
    "babel-plugin-react-css-modules": "3.3.3",
    "babel-plugin-transform-class-properties": "6.24.1",
    "babel-plugin-transform-decorators-legacy": "1.3.4",
    "babel-plugin-transform-object-rest-spread": "6.26.0",
    "babel-plugin-transform-runtime": "6.23.0",
    "babel-preset-env": "1.6.1",
    "babel-preset-react": "6.24.1",
    "enzyme": "3.3.0",
    "enzyme-adapter-react-16": "1.1.1",
    "eslint": "^4.19.1",
    "jest": "^22.4.3",
    "jest-css-modules": "1.1.0",
    "react-dom": "16.3.1",
    "react-test-renderer": "16.3.1"
  },
  "dependencies": {
    "@rentpath/react-ui-utils": "4.0.1",
    "babel-runtime": "6.26.0",
    "classnames": "2.2.5",
    "form-serialize": "0.7.2",
    "prop-types": "^15.6.1",
    "react-basic-datepicker": "rentpath/react-basic-datepicker",
    "react-image-gallery": "rentpath/react-image-gallery",
    "react-input-range": "1.3.0",
    "react-lazyload": "2.3.0",
    "react-motion": "^0.5.2",
    "react-spinners": "^0.2.6",
    "react-themed": "3.2.1"
  },
  "peerDependencies": {
    "lodash": "^4.17.5",
    "react": "^16.3.1",
    "react-dom": "^16.3.1"
  }
}
