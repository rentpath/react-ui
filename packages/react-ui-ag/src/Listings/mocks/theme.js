import { keyMirror } from '@rentpath/react-ui-utils'

export default keyMirror([
  'Carousel',
  'MobileMapListing',
  'ToggleButton',
  'RatingBar',
  'Button',
])
