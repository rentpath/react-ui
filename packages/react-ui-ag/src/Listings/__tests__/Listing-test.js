import React from 'react'
import { shallow, mount } from 'enzyme'
import { omit } from 'lodash'
import { Button, ToggleButton, ListingComponents } from '@rentpath/react-ui-core'
import ThemedListing from '../Listing'
import theme from './mocks/theme'

const Listing = ThemedListing.WrappedComponent

const baseListing = {
  bedrooms: '1-3 Bedrooms',
  bathrooms: '2 Bathrooms',
  name: 'Awesome Property!',
  city: 'Great Town',
  state: 'YA',
  price: '$1170+',
  phone: '678-907-1428',
  rating: {
    score: 4,
    label: '20',
  },
  photos: [
    {
      path: '',
      caption: '',
    },
    {
      path: '',
      caption: '',
    },
    {
      path: '',
      caption: '',
    },
  ],
}

const props = {
  listing: baseListing,
  theme,
  onClick: () => { },
  navigation: {
    next: {
      children: 'Next',
    },
    previous: {
      children: 'Previous',
    },
  },
  ctaButtons: [
    {
      valueLocation: 'phone',
      onClick: () => { },
      className: 'phone',
    },
    {
      children: 'send an email',
      onClick: () => { },
      className: 'contact',
    },
  ],
  favoriteButton: {
    onClick: () => { },
    children: 'â™¥',
  },
  photos: {
    server: 'https://image.rent.com/',
    dimensions: '280-120',
  },
  lazyLoad: false,
  listingInfoComponents: <div>Listing Info generated component</div>,
}

describe('ag/Listings/Listing', () => {
  describe('ctaButtons', () => {
    let wrapper
    let ctaClick
    let cardClick

    beforeEach(() => {
      ctaClick = jest.fn()
      cardClick = jest.fn()
      wrapper = shallow(
        <Listing
          {...props}
          ctaButtons={[{ children: 'foo', onClick: ctaClick }, { children: 'bar', onClick: ctaClick }]}
          onClick={cardClick}
          isActive
        />
      )
    })

    it('does not render a cta button if there is no text to display in the button', () => {
      wrapper = shallow(
        <Listing
          {...props}
          ctaButtons={[{ onClick: ctaClick }, { children: 'bar', onClick: ctaClick }]}
          onClick={cardClick}
          isActive
        />
      )
      expect(wrapper.find(Button).length).toEqual(1)
    })

    it('handle an array of cta buttons', () => {
      expect(wrapper.find(Button)).toHaveLength(2)
    })

    it('fire cta buttons on click action on click', () => {
      wrapper.find('[data-tid="cta-button"]').at(0).simulate('click')
      wrapper.find('[data-tid="cta-button"]').at(1).simulate('click')
      expect(ctaClick.mock.calls).toHaveLength(2)
    })

    it('fires cta buttons on click action when isActive is false', () => {
      wrapper = shallow(
        <Listing
          {...props}
          ctaButtons={[{ children: 'foo', onClick: ctaClick }, { children: 'bar', onClick: ctaClick }]}
          onClick={cardClick}
          isActive={false}
        />)
      wrapper.find('[data-tid="cta-button"]').at(0).simulate('click')
      wrapper.find('[data-tid="cta-button"]').at(1).simulate('click')
      expect(cardClick.mock.calls).toHaveLength(2)
    })
  })

  it('populates components from listingInfoComponents prop', () => {
    const wrapper = mount(
      <Listing
        {...props}
        listingInfoComponents={<div>inside listing info</div>}
      />
    )
    expect(wrapper.find('[data-tid="listing-info"]').at(0).text()).toEqual('inside listing info')
  })

  it('fires favoriteButton click action on favorite button click', () => {
    const favoriteClick = jest.fn()
    const wrapper = mount(
      <Listing
        {...props}
        favoriteButton={{ onClick: favoriteClick }}
      />
    )
    wrapper.find(ToggleButton).simulate('click')
    expect(favoriteClick).toHaveBeenCalledWith(baseListing, true)
  })

  it('does not fire cardClick on favorite button click', () => {
    const cardClick = jest.fn()
    const wrapper = mount(
      <Listing
        {...props}
        onClick={cardClick}
      />
    )
    wrapper.find(ToggleButton).simulate('click')
    expect(cardClick).not.toHaveBeenCalled()
  })

  it('does not fire cardClick on carousel nav click', () => {
    const cardClick = jest.fn()
    const wrapper = mount(
      <Listing
        {...props}
        onClick={cardClick}
      />
    )

    wrapper
      .find('[data-tid="carousel"]')
      .find('[className="image-gallery-left-nav"]')
      .simulate('click')
    expect(cardClick).not.toHaveBeenCalled()
  })

  it('does not fire cardClick on cta click', () => {
    const cardClick = jest.fn()
    const wrapper = mount(
      <Listing
        {...props}
        onClick={cardClick}
      />
    )
    wrapper.find('[data-tid="cta-button"]').at(0).simulate('click')
    expect(cardClick).not.toHaveBeenCalled()
  })

  it('fires onClick on card click', () => {
    const cardClick = jest.fn()
    const wrapper = mount(
      <Listing
        {...props}
        index={0}
        onClick={cardClick}
      />
    )
    wrapper.simulate('click')
    expect(cardClick).toHaveBeenCalled()
    expect(cardClick.mock.calls).toEqual([[0, baseListing]])
  })

  it('adds a inactive theme if the listing is inactive', () => {
    const wrapper = shallow(<Listing {...props} isActive={false} />)
    expect(wrapper.hasClass('Listing-inactive')).toBeTruthy()
  })

  it('adds a active theme if the listing is active', () => {
    const wrapper = shallow(<Listing {...props} isActive />)
    expect(wrapper.hasClass('Listing-active')).toBeTruthy()
  })

  it('sets the favorite button to favorited when the listing has isFavorited as true', () => {
    const favoritedProps = {
      ...props,
      listing: {
        ...props.listing,
        isFavorited: true,
      },
    }
    const wrapper = shallow(<Listing {...favoritedProps} />)
    expect(wrapper.find(ToggleButton).prop('value')).toBeTruthy()
  })

  it('sets the favorite button to not favorited when the listing has isFavorited as false', () => {
    const favoritedProps = {
      ...props,
      listing: {
        ...props.listing,
        isFavorited: false,
      },
    }
    const wrapper = shallow(<Listing {...favoritedProps} />)
    expect(wrapper.find(ToggleButton).prop('value')).toBeFalsy()
  })

  describe('photos', () => {
    it('renders a component with a PhotoCarousel if active', () => {
      const wrapper = shallow(<Listing {...props} />)

      expect(wrapper.find(ListingComponents.Photos)).toHaveLength(1)
      expect(wrapper.find(ListingComponents.Photo)).toHaveLength(1)
    })

    it('renders a component with one background image if inactive', () => {
      const wrapper = shallow(<Listing {...props} isActive={false} />)

      expect(wrapper.find(ListingComponents.Photos)).toHaveLength(0)
      expect(wrapper.find(ListingComponents.Photo)).toHaveLength(1)
    })

    it('renders a photo component even without a listing photos array', () => {
      const omittedPhotoProps = omit(props, 'photos')
      const wrapper = shallow(<Listing {...omittedPhotoProps} isActive={false} />)

      expect(wrapper.find(ListingComponents.Photos)).toHaveLength(0)
      expect(wrapper.find(ListingComponents.Photo)).toHaveLength(1)
    })
  })
})
