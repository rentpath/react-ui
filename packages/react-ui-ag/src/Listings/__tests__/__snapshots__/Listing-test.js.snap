// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ag/Listings/Listing photos renders a component with a PhotoCarousel with defaults if active 1`] = `
Object {
  "className": undefined,
  "lazyLoad": false,
  "lazyLoadGallery": false,
  "onSlide": [Function],
  "renderCustomControls": undefined,
  "server": "https://rentpath-res.cloudinary.com/",
  "showNav": true,
}
`;
